app/user.go

```diff
@@ -0,0 +1,597 @@
+		UserId:    user.Id,
+		RequestId: ctx.String(RequestIDFlagName),
+	}
+	resp, err := c.client.ResendUserInvite(c.ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to resend invitation for user: %w", err)
+	}
+	return PrintProto(resp.GetRequestStatus())
+}
+
+func (c *UserClient) deleteUser(
+	ctx *cli.Context,
+	userID string,
+	userEmail string,
+) error {
+	if userID != "" && userEmail != "" {
+		return fmt.Errorf("both user-id and user-email set")
+	}
+	if userID == "" && userEmail == "" {
+		return fmt.Errorf("both user-id and user-email cannot be empty")
+	}
+
+	res, err := c.client.GetUser(c.ctx, &authservice.GetUserRequest{
+		UserEmail: userEmail,
+		UserId:    userID,
+	})
+	if err != nil {
+		return err
+	}
+	req := &authservice.DeleteUserRequest{
+		UserId:          res.User.Id,
+		ResourceVersion: ctx.String(ResourceVersionFlagName),
+		RequestId:       ctx.String(RequestIDFlagName),
+	}
+	if req.ResourceVersion == "" {
+		req.ResourceVersion = res.User.ResourceVersion
+	}
+	resp, err := c.client.DeleteUser(c.ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to update user: %w", err)
+	}
+	return PrintProto(resp.GetRequestStatus())
+}
+
+func (c *UserClient) performUpdate(ctx *cli.Context, user *auth.User) error {
+	req := &authservice.UpdateUserRequest{
+		UserId:          user.Id,
+		Spec:            user.Spec,
+		ResourceVersion: ctx.String(ResourceVersionFlagName),
+		RequestId:       ctx.String(RequestIDFlagName),
+	}
+	if req.ResourceVersion == "" {
+		req.ResourceVersion = user.ResourceVersion
+	}
+	resp, err := c.client.UpdateUser(c.ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to update user: %w", err)
+	}
+	return PrintProto(resp.GetRequestStatus())
+}
+
+func (c *UserClient) setAccountRole(
+	ctx *cli.Context,
+	userID string,
+	userEmail string,
+	accountRole string,
+) error {
+	user, userRoles, err := c.getUserAndRoles(userID, userEmail)
+	if err != nil {
+		return err
+	}
+	accountRoleToSet, err := getAccountRole(c.ctx, c.client, accountRole)
+	if err != nil {
+		return err
+	}
+	for i := range userRoles {
+		if userRoles[i].Id == accountRoleToSet.Id {
+			return fmt.Errorf("user already has '%s' account role", accountRoleToSet.Id)
+		}
+	}
+	// check if this is the global admin role, and replace all existing roles
+	if accountRoleToSet.Spec.AccountRole.ActionGroup == auth.ACCOUNT_ACTION_GROUP_ADMIN {
+		// set the user account admin role
+		y, err := ConfirmPrompt(ctx, "Setting admin role on user, please confirm")
+		if err != nil || !y {
+			return err
+		}
+		userRoles = []*auth.Role{accountRoleToSet}
+	} else {
+		for i := range userRoles {
+			ur := userRoles[i]
+			if ur.Type != auth.ROLE_TYPE_PREDEFINED {
+				continue
+			}
+			// find the current admin role to replace
+			ar := ur.Spec.AccountRole
+			if ar != nil && ar.ActionGroup != auth.ACCOUNT_ACTION_GROUP_UNSPECIFIED {
+				// only swap the current admin role
+				userRoles[i] = accountRoleToSet
```

Comment:
should we break after this assignment?
also is there any scenario where we exit the for loop without doing any update at all? should we error if that happens?

Reply:
also whats the 2nd check for? in case there is a future admin role we add that the CLI doesn't understand? But in that case, then CLI cannot be used to update admin role since it can never replace it?